worker/src/handlers/importPersonas/importPersonaHandler.js

```diff
@@ -98,28 +101,33 @@ export default personaService => async ({
   organisation
 }, done) => {
   // establish lock
-  const lock = await establishLock({ structure, data, organisation });
-  if (!lock) {
-    await promisify(publish)({
-      queueName: PERSONA_IMPORT_QUEUE,
-      payload: {
-        index,
-        data,
-        personaImportId,
-        structure,
-        organisation
-      }
-    });
-    done();
-    return;
-  }
+  let lock;
+  const afterValidationCb = async () => {
+    lock = await establishLock({ structure, data, organisation });
+    if (!lock) {
+      await promisify(publish)({
+        queueName: PERSONA_IMPORT_QUEUE,
+        payload: {
+          index,
+          data,
+          personaImportId,
+          structure,
+          organisation
+        }
+      });
+      done();
+      return false;
+    }
+    return true;
+  };
 
   // Do the magic
   const { processedCount, totalCount } = await importPersona({
```

Comment:
Currently this will throw an exception because `importPersona` can now return `undefined` instead of an object.